runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/HttpProblem.java

```diff
@@ -97,8 +95,13 @@ public String getTitle() {
         return this.title;
     }
 
+    public int getStatusCode() {
+        return this.statusCode;
+    }
+
+    @Deprecated(forRemoval = true)
```

Comment:
If you `@Deprecate` something it is best practice to document this in JavaDocs what part of new code replaces deprecated part (even if it is in same class).

I'd suggest also that we should mark this - in JavaDocs - for removal even for next minor version (`3.2.0`). 
In case of community having struggle migrating to new version we can always revert it in `3.2.1`.

On the other hand would like to get your point of view why we cannot keep it for developer convinence.

Reply:
Good points. I wanted to get rid of this class from `HttpProblem entirely, but it's still heavily used in the builder by mappers. On the other hand, it will return null for 422 and other non-standard codes.

I'll remove this deprecation now, it's not that critical.

Reply:
Well, with this note on `null` statuses I'm actually in favor of removing it in future.

But agree on not being critical for release.